sdjournal/journal.go

```diff
@@ -84,13 +260,49 @@ func (m *Match) String() string {
 	return m.Field + "=" + m.Value
 }
 
+func (j *Journal) getFunction(name string) (unsafe.Pointer, error) {
+	j.mu.Lock()
+	f, ok := libsystemdFunctions[name]
+	if !ok {
+		var err error
+		f, err = j.lib.GetSymbolPointer(name)
+		if err != nil {
+			return nil, err
+		}
+
+		libsystemdFunctions[name] = f
+	}
+	j.mu.Unlock()
```

Comment:
Just defer it instead? Otherwise if there is an error when getting a symbol pointer, this will not be unlocked.